### PR TITLE
[Outreachy] Dashboard Redesign URL

### LINKED ISSUE #8831: Create Another Dashboard URL
This issue is part of #8513

It will fix:

- [ ] Create the new 'Dashboard' URL `/dashboard2`

### PR DESCRIPTION
**Pull request by RuthNjeri**
Fixes #8831 

I haven't fully completed the queries needed for the new dashboard page.

I would love some clarification.. Please have a look at the Dashboard screenshot on #8513 to get an idea on some of the questions I have raised within the comments. 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [X] PR is descriptively titled 📑 and links the original issue above 🔗
* [X] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [X] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -47,6 +47,29 @@ def dashboard
     end
   end
 
+  def dashboard2
+    # The new dashboard displays the blog and topics list
+    if current_user
+      @blog = Tag.find_nodes_by_type('blog', 'note', 1).first
+      # Topics a user has subscribed to
+      subscibed_tags = TagSelection.includes(:tag)
```
Comment (jywarren) [inline app/controllers/home_controller.rb:55]: This is looking like a great start, and i like that you're starting with just one query. I bet there are other pages we can look at to find fast standard ways to get this info, too! Here, I think we could use `user.subscriptions` -- 

https://github.com/publiclab/plots2/blob/6505f539cabcb9aa4eb9fc5c00af6cf9584351a2/app/models/user.rb#L191-L196

Then, i search for where this is used in code, to see examples:

https://github.com/publiclab/plots2/search?q=user.subscriptions - and indeed we see a nice example, which also includes the `tag` table, so we get the whole record we need. (`tag_selection` is just a join table, i think!)

```ruby
    <% current_user.subscriptions(:tag).includes(:tag).each do |subscription| %>
```

Actually i wish we had a unit test, as that's usually a great guide. So I'll check there too - aww, looks like https://github.com/publiclab/plots2/blob/main/test/unit/user_test.rb has no example for `user.subscriptions` -- that's too bad! Maybe we make a to-do for ourselves to add a simple unit test there.

As to many of the queries below, the template we'll be using actually does a lot of that work -- it expects a `@tags` variable and will build the whole listing from that:

https://github.com/publiclab/plots2/blob/main/app/views/tag/_topicCard.html.erb

So, if we do this, we might get very close!

```suggestion
      @tags = current_user.subscriptions(:tag).includes(:tag)
```

What do you think?

File: app/controllers/home_controller.rb [hunk 2]
```diff
@@ -47,6 +47,29 @@ def dashboard
     end
   end
 
+  def dashboard2
```
Comment (Sagarpreet) [inline app/controllers/home_controller.rb:50]: Can we have better name `dashboard2 `?

File: config/routes.rb [hunk 3]
```diff
@@ -227,6 +227,7 @@
   get 'feed/liked' => 'notes#liked_rss'
 
   get 'dashboard' => 'home#dashboard'
+  get 'dashboard2' => 'home#dashboard2'
```
Comment (Sagarpreet) [inline config/routes.rb:230]: `v2/dashboard` ?
USing v2 we can document also what was missing in v1 maybe, what do you think?

Comment (RuthNjeri) [reply config/routes.rb:230]: Yes, this makes sense! I can use v2

File: app/controllers/home_controller.rb (hunk 2 above)
Comment (RuthNjeri) [reply app/controllers/home_controller.rb:50]: Yeah, I can use v2 which sounds much better...

File: app/controllers/home_controller.rb (hunk 1 above)
Comment (RuthNjeri) [reply app/controllers/home_controller.rb:55]: This totally makes sense, thanks @jywarren, I'll do some further research and update the PR..

Comment (RuthNjeri) [reply app/controllers/home_controller.rb]: Hi, @jywarren I've added the subscriptions test as well as the query.

File: app/controllers/home_controller.rb [hunk 4]
```diff
@@ -47,6 +47,17 @@ def dashboard
     end
   end
 
+  def dashboard_v2
+    # The new dashboard displays the blog and topics list
+    if current_user
+      @blog = Tag.find_nodes_by_type('blog', 'note', 1).first
```
Comment (jywarren) [inline app/controllers/home_controller.rb:53]: Here we could add .limit(1) to optimize the query a bit.
```suggestion
      @blog = Tag.find_nodes_by_type('blog', 'note', 1).limit(1).first
```

**Review by jywarren — CHANGES_REQUESTED** [review]
This looks great! Are you interested in adding the partial template to display your list of topics in this pr or are you planning that for a follow-up? Awesome, Ruth!

File: app/controllers/home_controller.rb (hunk 4 above)
Comment (RuthNjeri) [reply app/controllers/home_controller.rb:53]: Alright 👌🏾

**Comment by RuthNjeri** [comment]
> This looks great! Are you interested in adding the partial template to display your list of topics in this pr or are you planning that for a follow-up? Awesome, Ruth!

Thanks 😄 , I was thinking of doing that with the design task in a follow-up PR...

File: app/controllers/home_controller.rb (hunk 4 above)
Comment (RuthNjeri) [reply app/controllers/home_controller.rb]: Done!

**Comment by RuthNjeri** [comment]
@jywarren I'm curious, in the new next PR, should I create a new topics card partial or should I reuse the one within the tag views?

**Comment by RuthNjeri** [comment]
> @jywarren I'm curious, in the new next PR, should I create a new topics card partial or should I reuse the one within the tag views?

oh, I'll recreate another one, I have just noticed slight differences...

**Comment by jywarren** [comment]
I wonder though if you may want to use conditionals to re-use the existing one to preserve consistency? But I trust your judgement!

OK, this should be ready to merge then!

**Comment by jywarren** [comment]
Let's check it out on stable.publiclab.org once it publishes! Great work!!!! 🎉 🎉 🎉 🎉

**Pull request merged**